src/components/cylc/gscan/GScan.vue

```diff
@@ -316,11 +316,20 @@ export default {
      * (natural sort).
      */
     sortedWorkflows () {
+      // Create a new object with state.name => order.
+      const workflowStateOrder = Object.fromEntries(
+        WorkflowStateOrder
+          .map(([state, order]) => {
+            return [state.name, order]
+          })
+      )
       return [...this.filteredWorkflows].sort((left, right) => {
-        if (left.status !== right.status) {
-          const leftState = WorkflowState.enumValueOf(left.status.toUpperCase())
-          const rightState = WorkflowState.enumValueOf(right.status.toUpperCase())
-          return leftState.enumOrdinal - rightState.enumOrdinal
+        const leftStateOrder = workflowStateOrder[left.status]
+        const rightStateOrder = workflowStateOrder[right.status]
+        // Here we compare the order of states, not the states since some states
+        // are grouped together, like RUNNING, PAUSED, and STOPPING.
+        if (leftStateOrder !== rightStateOrder) {
+          return leftStateOrder - rightStateOrder
```

Comment:
One case in a unit test was failing because of this. It ended up with the incorrect order since it was finding workflow `e` as `PAUSED` and workflow `a` as `RUNNING` and returning `0` since the states were different. When actually it was supposed to skip the states comparison since they have the same order now, and then sort alphabetically. :neckbeard: